src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts

```diff
@@ -207,12 +207,22 @@ function check_entity_property_usage(
       break;
 
     case "int":
+      if ((typeof value === "number" && Number.isInteger(value)) || typeof value === "string") return;
+
+      diagnoser.add(
+        `${parent}/${name}/${value}`,
+        `Property value is not a integer: ${value}`,
+        DiagnosticSeverity.error,
+        `behaviorpack.entity.property.${definition.type}.value`
+      );
+      break;
+
     case "float":
-      if (typeof value === "number" || typeof value === "string") return;
+      if ((typeof value === "number" && !Number.isInteger(value)) || typeof value === "string") return;
```

Comment:
Im not sure for this extra checks on floats 🤔 

cause 3 and 3.2 are still both valid floats

Reply:
I mean you can test it in game if you'd like; they've carried this over to items too in 1.21.40.
> Numbers and booleans are no longer interchangeable in the JSON input
> Floating point numbers are no longer accepted where an integer is expected

Reply:
Thats says floats are not accepted as integer. My comment is about integers being used in floats

Reply:
Vice versa is true for actor properties. You can test it yourself if you'd like
![image](https://github.com/user-attachments/assets/514a845e-6dca-48f2-8b53-6cc7df9aa223)

Reply:
What I mean is the following:

```json
"paeaefae:aeae": {
    "type": "float",
    "default": 3.2,
    "range": [ 0.0, 2.0 ]
}

"paeaefae:aeae": {
    "type": "float",
    "default": 3,
    "range": [ 0.0, 2.0 ]
}
```

both of these need to be valid, with your introduction of `!Number.IsInteger` the second case will fail

Reply:
@DaanV2

Reply:
Going to need to validate this myself in game, but I'm quite busy this week